plugins-ui/src/modules/review/components/leave-review/LeaveReview.tsx

```diff
@@ -6,26 +6,51 @@ import { useReviews } from "../../context/ReviewProvider";
 import StarContainer from "@/modules/shared/star-container/StartContainer";
 
 const LeaveReview = () => {
-  const { pluginId, addReview } = useReviews();
+  const { pluginId, addReview, reviewsMap } = useReviews();
 
   const [input, setInput] = useState("");
   const [rating, setRating] = useState(0);
   const [canReview, setCanReview] = useState(false);
+  const [isUpdating, setIsUpdating] = useState(false);
 
   useEffect(() => {
-    setCanReview(!!localStorage.getItem("authToken"));
-  }, []);
+    const authToken = localStorage.getItem("authToken");
+    const walletAddress = localStorage.getItem("walletAddress");
```

Comment:
I'd recommend avoiding storing `walletAddress` in `localStorage`, since the wallet connection state can change (e.g., the user may disconnect or switch accounts). Relying on `localStorage` could lead to stale or incorrect state. It’s generally better to track the connected address through a reactive source (like a wallet provider or global state), so the UI stays in sync with the actual wallet connection.

Reply:
i have created a WalletProvider and using that to manage wallet connect / disconnect functionality,